### PR TITLE
Use alternative meta fields

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44723

## Description
Added the -af flag for the ability to use the alternative fields as main fields in content items yml\json when creating content artifacts.

### CODE REVIEW

File: demisto_sdk/commands/common/content/objects/abstract_objects/general_object.py [hunk 1]
```diff
@@ -86,12 +95,10 @@ def dump(self, dest_dir: Optional[Union[Path, str]] = None) -> List[Path]:
...
             List[Path]: List of path created in given directory.
 
-        TODO:
-            1. Implement dump of modified object.
         """
-        dest_file = self._create_target_dump_dir(dest_dir) / self.normalize_file_name()
 
-        return [copyfile(src=self.path, dst=dest_file)]
+        dest_file = self._create_target_dump_dir(dest_dir) / self.normalize_file_name()
+        return [copyfile(src=self._path, dst=dest_file)]
```
Comment (yaakovi) [inline demisto_sdk/commands/common/content/objects/abstract_objects/general_object.py:101]: not sure what is the change here

File: demisto_sdk/commands/unify/yml_unifier.py [hunk 2]
```diff
@@ -89,8 +89,11 @@ def __init__(self, input: str, dir_name=INTEGRATIONS_DIR, output: str = '',
         self.ryaml.preserve_quotes = True
         self.ryaml.width = 50000  # make sure long lines will not break (relevant for code section)
         if self.yml_path:
-            with io.open(self.yml_path, 'r', encoding='utf8') as yml_file:
-                self.yml_data = self.ryaml.load(yml_file)
+            if yml_modified_data:
```
Comment (yaakovi) [inline demisto_sdk/commands/unify/yml_unifier.py:92]: replace the order of the conditions:
if modified, use the given value.
otherwise, check the path

**Review by yaakovi — COMMENTED** [review]
very nice

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -2319,3 +2320,44 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+    """
+    Go over all of the given content item fields and if there is a field with an alternative name, which is marked
+    by '_x2', use that value as the value of the original field (the corresponding one without the '_x2' suffix).
+    Args:
+        content_item: content item object
+
+    """
+    as_dict_types = (dict, CommentedMap)
+    current_dict = content_item.to_dict() if type(content_item) not in as_dict_types else content_item
+    copy_dict = current_dict.copy()  # for modifyting dict while iterating
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: this way the content item .yml will have these extra fields that will be misleading as name & name_x2 will have the same value. is not duplicating worth that?

File: demisto_sdk/commands/common/content/objects/abstract_objects/general_object.py (hunk 1 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/content/objects/abstract_objects/general_object.py:101]: Removed the 'TODO' as dumping modified object is now supported

File: demisto_sdk/commands/common/tools.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply demisto_sdk/commands/common/tools.py:2415]: True, but it will only be this way in the bucket.
Your call.

**Comment by bakatzir** [comment]
mentioning that this(or next pr) is a prequisite for https://github.com/demisto/etc/issues/45414

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+
+    Return (list of found { 'id': command name, 'source': command source }, list of found script and playbook names)
+    """
+    commands = []
+    detailed_commands = []
+    scripts_and_pbs = []
+    if file_type == FileType.TEST_PLAYBOOK or file_type == FileType.PLAYBOOK:
+        tasks = data.get('tasks')
+        for task_num in tasks.keys():
+            task = tasks[task_num]
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

File: demisto_sdk/commands/common/tools.py [hunk 5]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+            inner_task = task.get('task')
+            task_type = task.get('type')
+            if inner_task and task_type == 'regular' or task_type == 'playbook':
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

File: demisto_sdk/commands/common/tools.py [hunk 6]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+    Args:
+        data: The yml data as extracted with get_yaml
+        file_type: The FileType of the data provided.
+
+    Return (list of found { 'id': command name, 'source': command source }, list of found script and playbook names)
+    """
+    commands = []
+    detailed_commands = []
+    scripts_and_pbs = []
+    if file_type == FileType.TEST_PLAYBOOK or file_type == FileType.PLAYBOOK:
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

File: demisto_sdk/commands/common/tools.py [hunk 7]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+            commands.append(integration_command.get('name'))
+
+    for command in commands:
+        command_parts = command.split('|||')
+        if len(command_parts) == 2:
+            detailed_commands.append({
+                'id': command_parts[1],
+                'source': command_parts[0]
+            })
+        else:
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: 'source': command_parts[0]

File: demisto_sdk/commands/common/tools.py [hunk 8]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+                else:
+                    if task_type == 'playbook':
+                        scripts_and_pbs.append(inner_task.get('playbookName'))
+                    elif inner_task.get('scriptName'):
+                        scripts_and_pbs.append(inner_task.get('scriptName'))
+        if file_type == FileType.PLAYBOOK:
+            playbook_id = get_entity_id_by_entity_type(data, PLAYBOOKS_DIR)
+            scripts_and_pbs.append(playbook_id)
+
+    if file_type == FileType.SCRIPT:
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

File: demisto_sdk/commands/common/tools.py [hunk 9]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+        script_id = get_entity_id_by_entity_type(data, SCRIPTS_DIR)
+        scripts_and_pbs = [script_id]
+        if data.get('dependson'):
+            commands = data.get('dependson').get('must', [])
+
+    if file_type == FileType.INTEGRATION:
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md [hunk 10]
```diff
@@ -59,4 +59,58 @@ If you are interested in contributing, visit our contribution process [here](htt
 
 Thank you for contributing to Cortex XSOAR.
 
+If you are interested in contributing, visit our contribution process [here](https://xsoar.pan.dev/docs/contributing/contributing).#### This pack was co-authored by:
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:62]: Done

File: demisto_sdk/commands/common/tools.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py:2415]: This way makes more sense to me, so leaving as is

File: demisto_sdk/commands/common/tools.py [hunk 11]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
...
+        for task_num in tasks.keys():
+            task = tasks[task_num]
+            inner_task = task.get('task')
+            task_type = task.get('type')
+            if inner_task and task_type == 'regular' or task_type == 'playbook':
+                if inner_task.get('iscommand'):
+                    commands.append(inner_task.get('script'))
+                else:
+                    if task_type == 'playbook':
+                        scripts_and_pbs.append(inner_task.get('playbookName'))
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

File: demisto_sdk/commands/common/tools.py [hunk 12]
```diff
@@ -2341,6 +2341,66 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
     return list(set(using_integrations_pkg_paths + using_scripts_pkg_paths))
 
 
+def listdir_fullpath(dir_name: str) -> List[str]:
+    return [os.path.join(dir_name, f) for f in os.listdir(dir_name)]
+
+
+def get_scripts_and_commands_from_yml_data(data, file_type):
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py]: As discussed, not part of my code

**Review by yaakovi — APPROVED** [review]

**Pull request merged**